apps/cms.c

```diff
@@ -1012,7 +1012,7 @@ int cms_main(int argc, char **argv)
 
             pctx = CMS_RecipientInfo_get0_pkey_ctx(ri);
             if (kparam != NULL) {
-                if (!cms_set_pkey_param(pctx, kparam->param))
+                if (pctx != NULL && !cms_set_pkey_param(pctx, kparam->param))
```

Comment:
IMO a better fix would be to check pctx for being != NULL immediately after obtaining it from CMS_RecipientInfo_get0_pkey_ctx() .

Reply:
Right.